### PR TITLE
Sbt 1.0 Support

### PR DESCRIPTION
**Pull request by godenji**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  Seq(
+    organization := "com.typesafe.sbteclipse",
+    sbtPlugin := true,
+    git.baseVersion := baseVersion,
+    sbtVersion in GlobalScope := {
+      System.getProperty("sbt.build.version", (sbtVersion in GlobalScope).value)
+    },
+    scalacOptions ++= Seq("-unchecked", "-deprecation",
+      CrossVersion.partialVersion(scalaVersion.value) match {
+        case Some((2, 12)) => "-target:jvm-1.8"
```
Comment (benmccann) [inline build.sbt]: I'd be fine with only supporting Java 8 regardless of Scala version

**Comment by benmccann** [comment]
Do we really have to have two copies of every scripted test file? It seems rather frustrating that we'd have to make every change to our tests in two places from now on. I'd rather just test one version of the plugin that duplicate every file

File: build.sbt (hunk 1 above)
Comment (godenji) [inline build.sbt]: Scala 2.10, which sbt 0.13 is based on, doesn't work with Java 8.

**Comment by godenji** [comment]
> Do we really have to have two copies of every scripted test file?

The end game is to ditch `0.13` entirely, so not sure it's worth the bother to refactor the tests. Future releases could, for example, require sbt 1.0+; at that point we just move everything back under `src/main/scala` and `src/sbt-test`.

Travis is failing, looks like need to exclude 2.10 from Java 8 phase of build.

**Comment by benmccann** [comment]
We still have two copies of every file. Can we delete the `.sbt-test-0.13` directory?

It probably would be helpful to leave one of the most basic tests for `sbt-test-0.13` so that we can make sure we don't totally break 0.13

**Comment by godenji** [comment]
> We still have two copies of every file. Can we delete the .sbt-test-0.13 directory?

Sure.

> It probably would be helpful to leave one of the most basic tests for sbt-test-0.13 so that we can make sure we don't totally break 0.13

Travis build currently only tests against 1.0/2.12. Of course, this PR changes nothing wrt to 0.13 sources (just moves `Eclipse.scala` and `EclipsePlugin.scala` from `src/main/scala` to `src/main/scala-sbt-0.13`), so for 0.13 users this will be a non-release.

I'll keep 0.13 tests in place locally, `^scripted` passes for both sbt versions.

Let me know what you want to do, would be nice to get a release out the door.

**Comment by benmccann** [comment]
The ideal situation would be to have Travis test everything against both 1.0 and 0.13 without duplicating any code. Is that not possible?

If it's not possible to run against two versions of SBT without duplicated code, can we have Travis test everything against 1.0 and run a single test against 0.13? (so that we can ensure 0.13 at least somewhat runs without having to duplicate code)

It it's not possible to run two versions of SBT on Travis, then can we have it so that it runs the tests against both versions locally as you suggested?

**Comment by godenji** [comment]
> The ideal situation would be to have Travis test everything against both 1.0 and 0.13 without duplicating any code. Is that not possible?

Almost, `02-contents` fails for 0.13, so at minimum we would need to duplicate those tests (or do a run time check on sbt version and patch tests accordingly).

Currently `sbtTestDirectory` is set based on `sbtVersion`; if I can fix `02-contents` tests then I should be able to merge the tests under a single directory and avoid duplication.

Have to get Travis working as well, previous attempt failed due to Travis running tests against environment settings that won't work (2.12 + jdk7 and 2.10 + jdk8).

I'll update the PR tomorrow (good questions, btw, should clean things up quite a bit).

**Comment by benmccann** [comment]
Ok, merging under a single directory definitely sounds like the way to go then. Thanks for taking this on!

**Comment by godenji** [comment]
@benmccann All set, tests merged, build passes -- looks like we're good to here.

**Comment by benmccann** [comment]
There's a `.bak` file on the PR that I'm guessing you don't want to check in

**Comment by godenji** [comment]
Thanks, totally missed that, `.bak` file removed.

**Comment by benmccann** [comment]
Awesome work on this! Thank you!

**Comment by benmccann** [comment]
I've released this as 5.2.1

**Pull request merged**